libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/ApolloResponse.kt

```diff
@@ -71,10 +83,10 @@ private constructor(
   @get:JvmName("dataAssertNoErrors")
   val dataAssertNoErrors: D
```

Comment:
Where do we stand on this API? Since we're moving most of the API to be non-throwing, should we deprecate it? And maybe offer `Result` like helpers instead?

Reply:
Also I think there's a soft consensus that `val` should not throw.

Reply:
Yeah, the goal was to help centralizing error handling, but now you have it anyway by checking `.errors` and `.exception` without a need to throw. Ok to deprecate it! 🧹

Reply:
As discussed, let's wait a bit before we deprecate

Reply:
Reverted it and opened #4696